base/applications/notepad/dialog.c

```diff
@@ -332,48 +333,46 @@ GetPrintingRect(HDC hdc, RECT margins)
 
 static BOOL DoSaveFile(VOID)
 {
-    BOOL bRet = TRUE;
+    BOOL bRet = FALSE;
     HANDLE hFile;
-    LPTSTR pTemp;
-    DWORD size;
+    DWORD cchText;
 
-    hFile = CreateFile(Globals.szFileName, GENERIC_WRITE, FILE_SHARE_WRITE,
-                       NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
-    if(hFile == INVALID_HANDLE_VALUE)
+    hFile = CreateFileW(Globals.szFileName, GENERIC_WRITE, FILE_SHARE_WRITE,
+                        NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (hFile == INVALID_HANDLE_VALUE)
     {
         ShowLastError();
         return FALSE;
     }
 
-    size = GetWindowTextLength(Globals.hEdit) + 1;
-    pTemp = HeapAlloc(GetProcessHeap(), 0, size * sizeof(*pTemp));
-    if (!pTemp)
+    cchText = GetWindowTextLengthW(Globals.hEdit);
+    if (cchText <= 0)
     {
-        CloseHandle(hFile);
-        ShowLastError();
-        return FALSE;
+        bRet = TRUE;
     }
-    size = GetWindowText(Globals.hEdit, pTemp, size);
-
-    if (size)
+    else
     {
-        if (!WriteText(hFile, (LPWSTR)pTemp, size, Globals.encFile, Globals.iEoln))
```

Comment:
The old code fails at `WriteText` due to emptiness.

Reply:
https://github.com/reactos/reactos/blob/5913e5bd6b9e41fd3fa77de923c94d8d5d43b92d/base/applications/notepad/text.c#L361

Reply:
`iRequiredBytes` was zero for empty files.